src/data.js

```diff
@@ -141,33 +143,67 @@ Namespace.ensureNamespace = function(namespace, expires) {
 };
 
 /**Delete expired entries */
-Namespace.expireEntries = function(parent, continuationToken) {
-
+Namespace.expireEntries = function(parent, continuationToken=null) {
+  console.log('parent ->', parent);
+  
   return this.query({
     parent: parent,
   },
   {
     limit:         500,
     continuation:   continuationToken,
-  }).then(async (data) => {
+  }).then(async (data) => { 
+    console.log(data);
     var dataLength = data.entries.length;
     
     for (var i=0; i<dataLength; i++) {
       let entry = data.entries[i];
       let namespace = parent + '.' + entry.name;
-      await Namespace.expireEntries(namespace) ;
+      if (parent.length === 0 || entry.name.length === 0) {
+        namespace = parent + entry.name;
+      }
+      console.log ('namespace ->', namespace);
+      await Namespace.expireEntries(namespace);
 
       // insertTask is careful to update expires of entries 
       // from the root out to the leaf. A parent's expires is
       // always later than the child's. Hence, we can delete an
       // entry without checking its children.
+      console.log('expires ->', entry.expires.getTime());
+      console.log('current_time ->', Date.now());
       if (entry.expires.getTime() < Date.now()) {
         entry.remove(false, true);
       }
-    }            
+      await IndexedTask.ExpireTasks(namespace);
```

Comment:
I think this needs to happen before the entry itself is removed (where Namespace.expireEntries is called).